zebra-rpc/src/methods.rs

```diff
@@ -1116,73 +1149,80 @@ where
         .boxed()
     }
 
-    // TODO: use HexData or SentTransactionHash to handle the transaction ID
     fn get_raw_transaction(
         &self,
-        txid_hex: String,
+        HexData(txid): HexData,
```

Comment:
Thanks, I did that in [bdb8372](https://github.com/ZcashFoundation/zebra/pull/9049/commits/bdb837284d7fe8089b25e19d180b68d82a2b3ef6). The error code is not compatible with `zcashd`, but it wasn't anyway if the submitted string wasn't hex. I'll try to figure out if we need compatibility here and open another PR if we do.

Reply:
I returned to using `String`s for the TXIDs even though they provide a weaker semantic abstraction. The reason is that `String`s are exactly what the RPC gets from the client. This means we have greater test coverage (and error control). Using `transaction::Hash` felt great, but then I had to submit that directly to the RPC instead of a `String`, and we couldn't test deserialization properly, which led to bugs like https://github.com/ZcashFoundation/zebra/pull/9049#issuecomment-2536303946.